src/api/models/ActionModel.js

```diff
@@ -1,28 +1,28 @@
 const mongo = require("mongoose");
+const uniqueValidator = require("mongoose-unique-validator");
 
 const ActionSchema = new mongo.Schema({
-  name: { type: String, required: true },
+  name: { type: String, required: true, unique: true },
   description: { type: String },
-  categories: { type: [mongo.Schema.Types.ObjectId] },
+  categories: [{ type: mongo.Schema.Types.ObjectId, ref: "Category" }],
   location: {
-    type: {
+    name: {
       type: String,
-      enum: ["Point"],
-      required: true
+      required: true,
```

Comment:
If the fields `location.name` & `location.coordinates` are required, it means that an action cannot be created without a location.

In the case that this is the intended behavior, you should add an error message in the ActionController as well.

Reply:
Am I not doing this already?

Reply:
You have an if statement for `req.body.location` but no else with an error.